README.md

```diff
@@ -1,66 +1,141 @@
+### On Initialized
+The plugin is successfully initialized. Triggered by the [Set Token](#set-token) function.
 
-It is also not difficult to extend plugin functionality by just knowing that:
-- ADolbyIoConference is the interface between the Unreal Engine and the SDK
-- FSdkAccess is the class where most of the calls to the SDK are made
+When this event is received, the plugin is ready for use. You can now, for example, call this Blueprint's [Connect](#connect) function. Once connected, the [On Connected](#on-connected) event will trigger.
 
-Therefore, the typical workflow to provide new features is to extend ADolbyIoConference's API and implement the functionality in FSdkAccess.
+Example:  
+![example](Images/on_initialized.PNG)
+
+### On Connected
+Successfully connected to conference. Triggered by the [Connect](#connect) function. Provides the ID of the local participant in its argument.
+
+Example:  
+![example](Images/on_connected.PNG)
+
+### On Disconnected
+Disconnected from conference. Triggered when disconnected by any means (in particular by the [Disconnect](#disconnect) function).
+
+Example:  
+![example](Images/on_disconnected.PNG)
+
+### On Remote Participants Changed
+Triggered when participants are added to or removed from the conference. Provides the IDs of the remote participants in its argument.
+
+Example:  
+![example](Images/on_remote_participants_changed.PNG)
+
+### On Active Speakers Changed
+Triggered when participants start or stop speaking. Provides the IDs of the current speakers in its argument.
+
+Example:  
+![example](Images/on_active_speakers_changed.PNG)
+
+### On Audio Levels Changed
+There are new audio levels available. Provides a string-to-float mapping of participant IDs to their audio levels (0.0 is silent, 1.0 is loudest) in its argument. Triggered by the [Get Audio Levels](#get-audio-levels) function.
+
+Example:  
+![example](Images/on_audio_levels_changed.PNG)
+
+## Functions
+### Set Token
+Initializes or refreshes the client access token. Takes the token as parameter. Initializes the plugin unless already initialized. Triggers [On Initialized](#on-initialized) if initialization was successful.
+
+For quick testing, you can manually obtain a token from the [Dolby.io dashboard](https://dashboard.dolby.io/) and paste it directly into the node.
+
+You may use the [Set Token Using Key and Secret](#set-token-using-key-and-secret) function instead for convenience during onboarding.
+
+Example:  
+![example](Images/on_token_needed.PNG)
+
+### Set Token Using Key and Secret
+Similar to [Set Token](#set-token) except it takes an app key and secret as parameters and automatically generates a token. **Do not use this function in production and do not allow your app key and secret to leak.**
```

Comment:
Recommended text:

For convenience during early development and prototyping, this function is provided to acquire a client access token directly from within the application. However, please note we do not recommend using this mechanism in the production software for [security best practices](https://docs.dolby.io/communications-apis/docs/guides-client-authentication). App secret needs to be protected and not to be included in the application.

Reply:
Will be applied in #27.